### PR TITLE
Ignore empty Sink

### LINKED ISSUE #48: Error reported when sink is not defined, although it's optional
The log level is "info", but the message still contains "error". We should remove this message since it's irrelevant and confusing.

Sample:

```json
{
  "level": "info",
  "ts": "2021-04-07T10:49:23.721Z",
  "logger": "controller.github.com-triggermesh-bumblebee-pkg-reconciler-controller.Reconciler",
  "caller": "controller/transformation.go:130",
  "msg": "Sink resolution error: destination missing Ref and URI, expected at least one",
  "commit": "8c552af",
  "knative.dev/traceid": "375da53f-cbbb-4bb8-86d0-2197b891e388",
  "knative.dev/key": "mynamespace/mytransformation"
}
```

### PR DESCRIPTION
**Pull request by tzununbekov**
If Sink is not set in the transformation Spec do not try to resolve it.
Closes #48.

### CODE REVIEW

File: pkg/reconciler/controller/transformation.go [hunk 1]
```diff
@@ -114,21 +114,12 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, trn *transformationv1alp
...
-			dest.Ref.Namespace = trn.GetNamespace()
-		}
-	}
-
 	var sink string
-	uri, err := r.sinkResolver.URIFromDestinationV1(ctx, *dest, trn)
-	if err != nil {
-		logger.Infof("Sink resolution error: %v", err)
-	} else {
+	if trn.Spec.Sink != (duckv1.Destination{}) {
```
Comment (antoineco) [inline pkg/reconciler/controller/transformation.go:118]: This comparison will not work because the struct's fields are pointers.

Comment (antoineco) [reply pkg/reconciler/controller/transformation.go:118]: I remove what I said, it's fine with `!=` if Ref and URI are just compared to `nil` 😅

**Review by antoineco — APPROVED** [review]

File: pkg/reconciler/controller/transformation.go (hunk 1 above)
Comment (tzununbekov) [reply pkg/reconciler/controller/transformation.go:118]: why? it is working both on the controller that I deployed manually and on the [playground](https://play.golang.org/p/wlycdAvOjz3)

Comment (tzununbekov) [reply pkg/reconciler/controller/transformation.go:118]: sorry, missed the second message

**Pull request merged**